tests/general_checks.py

```diff
@@ -37,6 +39,11 @@ def general_output_checks(
             output_data, expected_results, equal_nan=True, rtol=rtol
         )
 
+    if expected_dims:
```

Comment:
```suggestion
    if expected_dims is not None:
```
Could want to compare against an empty list